### ISSUE TITLE
How long did it take you to train on the MIMIC-CXR dataset?

### ISSUE THREAD

**Issue by min121101** [GaryGuTC/COMG_model#14]
Hello,

thank you for the great work and the great repo.

It takes about 50 hours per epoch for us to use your model training on the MIMIC-CXR dataset.

We used the same batch_size and num_worker as you, and the device is an NVIDIA 3090Ti.

Do you have other acceleration settings? I found that dataloader is very time-consuming when reading masks. Because of IO limitations, each iter needs to open more than 70 mask.jpg?

**Comment by GaryGuTC** [comment]
Hi 
1. Time-consuming: [here](https://github.com/GaryGuTC/COMG_model/issues/13#issuecomment-2097606992).
2. The 50 hours / epoch for 3090 Ti is too slow; please double-check.
3. I suggest that:
      1. Saving all mask information or part of them as `.npy` files. It can significantly improve speed.
      2. Try to do the majority of work on the IU-Xray dataset and final verification on the MIMIC-CXR dataset.
 
Best regards

**Comment by you22333** [comment]
I am also experiencing very slow training. Are there any solutions to this problem now?
